clippy_lints/src/else_if_without_else.rs

```diff
@@ -0,0 +1,71 @@
+//! lint on if expressions with an else if, but without a final else branch
+
+use rustc::lint::*;
+use syntax::ast::*;
+
+use utils::{in_external_macro, span_lint_and_sugg};
+
+/// **What it does:** Checks for usage of if expressions with an else if branch,
+/// but without a final `else` branch.
+///
+/// **Why is this bad?** Some coding guidelines require this.
+///
+/// **Known problems:** None.
+///
+/// **Example:**
+/// ```rust
+/// if x.is_positive() {
+///     a()
+/// } else if x.is_negative() {
+///     b()
+/// }
+/// ```
+///
+/// Could be written:
+///
+/// ```rust
+/// if x.is_positive() {
+///     a()
+/// } else if x.is_negative() {
+///     b()
+/// } else {
+///     // we don't care about zero
```

Comment:
zero is positive I think :D
just throw an `unreachable!()` here

Reply:
Actually, {integer}.is_positive() returns false for 0, which is what I had in mind.